File: Backend/ecommerce/src/main/java/com/luv2code/springbootecommerce/model/entity/Customer.java
package com.luv2code.springbootecommerce.model.entity;

import com.luv2code.springbootecommerce.enums.Role;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
public class Customer extends BaseEntity{

    private String firstName;

    private String lastName;

    private String email;

    private String phoneNumber;

    private String address;

    @Enumerated(EnumType.STRING)
    private Role role;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "customer")
    private Set<Order> orders = new HashSet<>();

    public void add(Order order){
        if(order != null){
            if(orders == null){
                orders = new HashSet<>();
            }
            orders.add(order);
            order.setCustomer(this);
        }
    }
}

File: Frontend/angular-ecommerce/app/common/customer.ts
import { Role } from "../enum/Role.enum";

export class Customer {

    constructor (
                public id: number,
                public firstName: string,
                public lastName: string,
                public email: string,
                public phoneNumber: string,
                public role: Role
                ) { }
    // id?: number;
    // firstName?: string;
    // lastName?: string;
    // email?: string;
    // phoneNumber?: string;
    // role?: Role;

}
